src/Symfony/Component/HttpKernel/Tests/HttpCache/ResponseCacheStrategyTest.php

```diff
@@ -370,13 +370,29 @@ public function cacheControlMergingProvider()
         ];
 
         yield 'merge max-age and s-maxage' => [
-            ['public' => true, 's-maxage' => '60', 'max-age' => null],
+            ['public' => true, 'max-age' => '60'],
             ['public' => true, 's-maxage' => 3600],
             [
                 ['public' => true, 'max-age' => 60],
             ],
         ];
 
+        yield 'subresponse without lifetime does not remove lifetime for main response' => [
```

Comment:
```suggestion
        yield 'public subresponse without lifetime does not remove lifetime for main response' => [
```

if the subresponse was private, the whole response would remove the lifetime and make the response private, right?

Reply:
Yes, at least that would me my intention. I need to double check this is in fact covered by a test.

Reply:
We have this: 

```php
        yield 'resolve to lowest possible max-age' => [
            /* expected */ ['public' => false, 'private' => true, 's-maxage' => false, 'max-age' => '60'],
            /* master */ ['public' => true, 'max-age' => 3600],
            [
                /* subrequest */ ['private' => true, 'max-age' => 60],
            ],
        ];
```

Does that resolve this comment @dbu?

Reply:
yep, it resolves the question. i'd still update the test description as per my suggestion to clarify that the important bit is that the subrequest is public.

Reply:
Did that :+1:.